### PR TITLE
Minimal metainfo API

### PR DESCRIPTION
**Pull request by mniewrzal**
Initial draft for minimal `metainfo` API

### CODE REVIEW

File: satellite/metainfo/metainfo.go [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+	mon          = monkit.Package()
+	segmentError = errs.Class("metainfo error")
+)
+
+// APIKeys is api keys store methods used by endpoint
+type APIKeys interface {
+	GetByKey(ctx context.Context, key console.APIKey) (*console.APIKeyInfo, error)
+}
+
+type Server struct {
```
Comment (egonelbre) [inline satellite/metainfo/metainfo.go]: Endpoint

File: satellite/metainfo/metainfo.go [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+	logger *zap.Logger
```
Comment (egonelbre) [inline satellite/metainfo/metainfo.go]: log

Comment (mniewrzal) [reply satellite/metainfo/metainfo.go]: ah yeah, wrong copy-paste

File: pkg/storage/segments/store.go [hunk 3]
```diff
@@ -418,3 +417,35 @@ func convertTime(ts *timestamp.Timestamp) time.Time {
 	}
 	return t
 }
+
+func split(path storj.Path) (projectID, bucket string, objectPath storj.Path, segmentIndex int64, err error) {
+	comp := storj.SplitPath(path)
+	if len(comp) < 3 {
```
Comment (mniewrzal) [inline pkg/storage/segments/store.go]: It is ok for accessing only bucket? Won't it be `l/bucket_name` without additional path? Also I'm not sure if `projectID` will be visible on client side.

**Comment by mniewrzal** [comment]
merged into `hammer` branch

**Pull request closed without merging**